Warn if user tries to login with API key from Cloud 1

anna-geller opened the pull request:
### Problem 

We've seen some users trying to log in to Cloud 2 with Cloud 1 API key. It would be great to direct them to the right Cloud URL.

### Example

![image](https://user-images.githubusercontent.com/86264395/192163542-92b5347d-de93-468d-88d1-c401c0a20d78.png)

### Checklist
<!-- These boxes may be checked after opening the pull request. -->

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests 
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`
  <!--  If you do not have permission to add a label, a maintainer will add one for you and check this box. -->

madkinsz commented on src/prefect/cli/cloud.py:
```diff
@@ -160,6 +160,10 @@ async def login(
                 for workspace in workspaces
             }
         except CloudUnauthorizedError:
+            if not key.startswith("pnu"):
+                exit_with_error(
+                    "Make sure you are using API key from https://app.prefect.cloud (Prefect 2) rather than https://cloud.prefect.io (Prefect 1)"
+                )
             exit_with_error(
                 "Unable to authenticate with Prefect Cloud. Please ensure your credentials are correct."
             )
```
Something in this style would make more sense to me:

```suggestion
            help_message = "Please ensure your credentials are correct."
            if not key.startswith("pnu"):
                help_message = "Your key is not in our expected format."
            exit_with_error(
                "Unable to authenticate with Prefect Cloud. {help_message}"
            )
```

It seems nice retain the Cloud 1 messaging, but with your implementation if I provide the key "foo" I'll get a message about Prefect 1 vs 2 which doesn't make much sense. Is there a way we can detect that the key is from Cloud 1 from the format?

anna-geller replied:
> if I provide the key "foo" I'll get a message about Prefect 1 vs 2 which doesn't make much sense

It does make sense because the right Cloud2 key should start with `pnu_` and `foo` doesn't

madkinsz replied:
It doesn't make sense to hint that the key was retrieved from Cloud 1 when it's just arbitrary text. This could be confusing. I'd prefer to have separate warnings for these cases:
- Warn that it's not in the expected format if it doesn't start with the expected Cloud 2 prefix
- Warn that it's from the wrong application (1 vs 2) if it actually looks like a Cloud 1 key

anna-geller replied:
thx a lot, 100% agree. Does this look good or would you change the wording/structure?

![image](https://user-images.githubusercontent.com/86264395/192163320-e304640b-ce18-4dfe-8271-8af1a217474c.png)

madkinsz commented:
Since this improves the UX without fixing a bug, I've updated this to an `enhancement`. This is also a great type of change to include an example for. We should also add a test covering whatever behavior we decide on.

anna-geller commented:
> Since this improves the UX without fixing a bug, I've updated this to an `enhancement`. This is also a great type of change to include an example for in the PR body. We should also add a test covering whatever behavior we decide on.

Thanks so much, great point, I updated the PR body
